text/orderer-v3.md

```diff
@@ -0,0 +1,394 @@
+In fact, by making the Fabric ordering service framework "import friendly" from external repositories,
+the various development teams of the open source community could implement their own consensus protocols that 
+fits their setting and integration with Fabric peers would not require changes in the Fabric codebase.
+
+**Proper responsibility for support**: One of the inhibitors for integrating new consensus protocols and libraries into Fabric 
+is that troubleshooting an issue requires knowledge of the consensus protocol and its implementation.
+Since with the proposed approach a new consensus protocol can be introduced to Hyperedger Fabric without influencing the codebase of the Fabric core,
+it will not be expected nor required from the Fabric core maintainers to troubleshoot or maintain ordering service implementations 
+that are not part of the official Fabric core repositories. This fact could expedite the emergence of new scientific and industrial consensus implementations 
+for Fabric.
+
+# Orderer-to-orderer authentication and communication
+[communication]: #orderer-to-orderer-authentication-and-communication
+
+### gRPC API changes
+
+The current [gRPC API](https://github.com/hyperledger/fabric-protos/blob/release-2.4/orderer/cluster.proto) that orderers use to communicate, declares the channel each message is routed to in the message itself:
+```
+// ConsensusRequest is a consensus specific message sent to a cluster member.
+message ConsensusRequest {
+    string channel = 1;
+    bytes payload = 2;
+    bytes metadata = 3;
+}
+```
+
+In the implementation of the communication API, gRPC streams never share channels and a message for channel *foo* will never be sent through 
+the same gRPC stream of channel *bar* and vice versa. 
+
+The implemented approach is inefficient as it makes the channel name to be sent over and over redundantly.
+We shall define a gRPC API that is similar to the existing one, but with the channel name sent only during authentication (the `Auth` message will contain the channel name):
+
+```
+// OrdererMessaging defines communication between orderers.
+service OrdererMessaging {
+    // Connect estabishes a one way stream between orderers
+    rpc Connect(stream Request) returns (Response);
+}
+
+
+// Request wraps a message that is sent to an orderer.
+message Request {
+    oneof payload {
+        // consensus_request is a consensus specific message.
+        ConsensusRequest consensus_request = 2;
+        // submit_request is a relay of a transaction.
+        SubmitRequest submit_request = 3;
+        // Auth authenticates the orderer that initiated the stream.
+        Auth auth = 4;
+    }
+}
+
+message Auth {
+ ...
+ string channel = 7;
+ ...
+}
+
+```
+
+The `ConsensusRequest` and `SubmitRequest` will be defined similarly as the [v2.x gRPC API](https://github.com/hyperledger/fabric-protos/blob/release-2.4/orderer/cluster.proto),
```

Comment:
I hope, we don't remove the `channel` field from message `ConsensusRequest` for backward compatibility, otherwise it requires the downtime for version upgrade.

Reply:
If we would need a downtime, it wouldn't be because of switching to a different protobuf schema.